### ISSUE TITLE
Cline panel disappears randomly

### ISSUE THREAD

**Issue by orangejon** [cline/cline#1093]
### What happened?

Cline was previously working fine but now it randomly goes blank. Hiding and revealing the Cline panel has no effect, but restarting Vscode fixes it. (at least temporarily)

<img width="1555" alt="Screenshot 2024-12-31 at 12 44 12" src="https://github.com/user-attachments/assets/a27ca304-64b1-40bb-a0ba-97b84861a2a5" />

It might be related to the Github codespace disconnecting and reconnecting, but it's difficult to be sure.

### Steps to reproduce

1. Use Cline as normal
2. The Cline panel goes blank
3.

### Operating System

MacOS Sonoma but connected to a Github Codespace (Ubuntu)

### Cline Version

v3.0.7

**Comment by orangejon** [comment]
When this bug happens, clicking any of the buttons creates a "not found" error notification in the bottom-right:

<img width="1555" alt="Screenshot 2024-12-31 at 13 01 15" src="https://github.com/user-attachments/assets/fffed263-7fc0-4fe5-a06e-db250b933d55" />

**Comment by orangejon** [comment]
It definitely seems to happen when the codespace disconnects. The codespace reconnects automatically and the terminal history is restored, but Cline stays blank.

**Comment by orangejon** [comment]
Can I offer a bounty to fix this? Cline works well when it's working, but my Codespace is disconnecting and reconnectting every few minutes currently so this bug makes Cline essentially unusable :(

**Comment by GLStephen** [comment]
@orangejon This is happening to me too. It seems to happen when cline is handling relatively large console results, like the output of a system conmand, but it does not require much. 100 lines or so.

**Comment by orangejon** [comment]
Yes, for me that's usually unit test results, I guess that's probably the
only 'large' amount of content on the Terminal. It also seems to happen
when the codespace disconnects and reconnects, but not every time -
possibly only when it was in the middle of something? I've been using it a
lot this weekend but haven't managed to pin it down yet. I just restart
Vscode (closing and reopening the window isn't enough).

**Comment by alexpgross** [comment]
Did you figure this out? Our codespace seems to be disconnecting randomly every few minutes and Cline goes blank, but eventually restores. Wonder why it disconnects.

**Comment by orangejon** [comment]
No, I never found any solution, it still happens regularly for me :(

Okay, I removed my User folder (in /Users/[username]/Library/Application
Support/Code on Mac) to see if that will make a difference.

**Comment by orangejon** [comment]
Removing my User folder didn't fix it. The codespace seems to disconnect
after running my unit tests, then when it reconnects Cline is blank.

**Comment by spences10** [comment]
I've been getting this issue the last two days now, weird, not looked into it anymore

opening in editor opens it up again sometimes but it's inconsistent

**Comment by spences10** [comment]
I just switched to VS Code and don't see the same issues as using it in Cursor, the Cursor team really have been stepping on their own dicks with the recent releases coming from them!

**Comment by spences10** [comment]
oh! No, it;s doing it in VS Code too

this was after I enabled a load of MCP servers, maybe a memory issue?

**Comment by GLStephen** [comment]
Are you running in codespaces? I have seen this and it seems to improve when I run in devcontainers on docker desktop. I suspect a time out of some sort with a backend of some sort.

**Comment by amotivv** [comment]
Experiencing this much worse today - mostly with longer model responses. Is there any way to remove portions of the response history? Otherwise I have to abandon the affected chat and start a new one.

**Comment by spences10** [comment]
I just disabled a load of vs code extensions I wasn't using then I've not had the problem since

**Comment by celestial-vault** [comment]
Thanks for the detailed report! This issue has been consolidated into #4352 which addresses comprehensive remote workspace support, including GitHub Codespaces.

The core gray screen memory issue was resolved in v3.16.1, but the codespace disconnection/reconnection problem you're experiencing is part of broader remote environment challenges that #4352 will address with improved state persistence and connection handling.

**Comment by amotivv** [comment]
The gray screen issue has started happening quite often again for me, especially during extended coding sessions in the same chat. Thankfully, restarting VSCode gets me back going.

**Comment by keeper86** [comment]
It happens for me too. Its a fresh install.
